dspMetadataGUI/util/metaDataSet.py

```diff
@@ -0,0 +1,1290 @@
+            # LATER: remove with next breaking change
+            g = self.generate_rdf_graph()
+        try:
+            val = self.validation_result
+        except Exception:
+            # LATER: remove with next breaking change
+            print('Warning: no cached validation result available. Performance may be decreased.')
+            val = self.validate_graph(g)[0]
+        if val:
+            overall = 'Valid'
+        else:
+            overall = 'Invalid'
+        invalid = 0
+        missing = 0
+        optional = 0
+        valid = 0
+        for p in self.get_all_properties():
+            v, _ = p.validate()
+            if v == Validity.INVALID_VALUE:
+                invalid += 1
+            elif v == Validity.REQUIRED_VALUE_MISSING:
+                missing += 1
+            elif v == Validity.OPTIONAL_VALUE_MISSING:
+                optional += 1
+            elif v == Validity.VALID:
+                valid += 1
+        return f"{overall}  --  {invalid + missing} Problems; {valid} Values"
+
+    def get_turtle(self) -> str:
+        """
+        Get the turtle representation of the data.
+
+        Returns:
+            str: a turtle serialization of the data.
+        """
+        try:
+            ttl = self.turtle
+            if not isinstance(ttl, str):
+                raise Exception('turtle was not a string')
+            if ttl:
+                return ttl
+            else:
+                raise Exception('no turtle found')
+        except Exception as e:
+            print('Warning: No turtle cached. Performance may be decreased.')
+            print(f'    (Exception: {e})')
+            # LATER: remove with next breaking change
+            g = self.generate_rdf_graph()
+            return g.serialize(format='turtle').decode("utf-8")
+
+
+class DataClass(ABC):
+    """
+    Abstract parent class of all classes holding data.
+    """
+
+    def get_metadataset(self) -> MetaDataSet:
+        """
+        Returns the `MetaDataSet` to which this class belongs
+
+        Returns:
+            MetaDataSet: The owner MetaDataSet
+        """
+        return self.meta
+
+    def add_rdf_to_graph(self, graph: Graph, typename: str):
+        """
+        Adds the RDF representation of this class to a graph
+
+        Args:
+            graph (Graph): the `rdflib.Graph` to which the triples should be added
+            typename (str): the type name of this class
+        """
+        iri = self.get_rdf_iri()
+        type = dsp_repo[typename]
+        graph.add((iri, RDF.type, type))
+        for prop in self.get_properties():
+            graph += prop.get_triples(iri)
+
+    def get_rdf_iri(self) -> URIRef:
+        """
+        Return the IRI of the object.
+
+        This can be called to use the object as subject of an RDF triple.
+
+        Returns:
+            URIRef: the IRI
+        """
+        shortcode = self.get_metadataset().project.shortcode.value
```

Comment:
Property `self.get_metadataset().project.shortcode` is of type str. Is it necessary to invoke `value`on it?

Reply:
no, `Metadataset.project.shortcode` is of type `Property`, so that call is necessary here